crates/node/scripts/install.js

```diff
@@ -0,0 +1,139 @@
+      if (arch === 'ia32') return '@tailwindcss/oxide-win32-ia32-msvc'
+      return '@tailwindcss/oxide-win32-x64-msvc'
+    case 'linux':
+      if (arch === 'x64') {
+        return libc === 'musl'
+          ? '@tailwindcss/oxide-linux-x64-musl'
+          : '@tailwindcss/oxide-linux-x64-gnu'
+      } else if (arch === 'arm64') {
+        return libc === 'musl'
+          ? '@tailwindcss/oxide-linux-arm64-musl'
+          : '@tailwindcss/oxide-linux-arm64-gnu'
+      } else if (arch === 'arm') {
+        return '@tailwindcss/oxide-linux-arm-gnueabihf'
+      }
+      break
+    case 'freebsd':
+      return '@tailwindcss/oxide-freebsd-x64'
+
+    case 'android':
+      return '@tailwindcss/oxide-android-arm64'
+    default:
+      return '@tailwindcss/oxide-wasm32-wasi'
+  }
+}
+
+function isPackageAvailable(packageName) {
+  try {
+    require.resolve(packageName)
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+// Extract all files from a tarball to a destination directory
+async function extractTarball(tarballStream, destDir) {
+  if (!fs.existsSync(destDir)) {
+    fs.mkdirSync(destDir, { recursive: true })
+  }
+
+  return new Promise((resolve, reject) => {
+    tarballStream
+      .pipe(extract({ cwd: destDir, strip: 1 }))
+      .on('error', (err) => reject(err))
+      .on('end', () => resolve())
+  })
+}
+
+async function downloadAndExtractBinary(packageName) {
+  let tarballUrl = `https://registry.npmjs.org/${packageName}/-/${packageName.replace('@tailwindcss/', '')}-${version}.tgz`
+  console.log(`Downloading ${tarballUrl}...`)
+
+  return new Promise((resolve) => {
+    https
+      .get(tarballUrl, (response) => {
+        if (response.statusCode === 302 || response.statusCode === 301) {
+          // Handle redirects
+          https.get(response.headers.location, handleResponse).on('error', (err) => {
+            console.error('Download error:', err)
+            resolve()
+          })
+          return
+        }
+
+        handleResponse(response)
+
+        async function handleResponse(response) {
+          try {
+            if (response.statusCode !== 200) {
+              throw new Error(`Download failed with status code: ${response.statusCode}`)
+            }
+
+            await extractTarball(
+              response,
+              path.join(__dirname, '..', 'node_modules', ...packageName.split('/')),
+            )
+            console.log(`Successfully downloaded and installed ${packageName}`)
+          } catch (error) {
+            console.error('Error during extraction:', error)
+            resolve()
+          } finally {
+            resolve()
+          }
+        }
+      })
+      .on('error', (err) => {
+        console.error('Download error:', err)
+        resolve()
+      })
+  })
+}
+
+async function main() {
+  // Don't run this script in the package source
+  if (fs.existsSync(path.join(__dirname, '..', 'build.rs'))) {
+    return
+  }
```

Comment:
I think we should wrap this in a try/catch. Some people run stuff from Docker in the `/` path, so `../build.rs` will very likely crash.

Also let's hope nobody has a `build.rs` file in the parent directory 😅 

```suggestion
  // Don't run this script in the package source
  try {
    if (fs.existsSync(path.join(__dirname, '..', 'build.rs'))) {
      return
    }
  } catch {
    return
  }
```

Reply:
wait but the script is run within the npm package in the `scripts/` folder so the parent is always there and should not be on the root or am I misunderstanding something?

Reply:
still doesn't hurt to wrap in try catch I assume

Reply:
Ah right, executed from the `scripts` folder, I interpreted it as root of the project. Makes sense!